File: Java/프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        Queue<Integer> q1 = new LinkedList<>();
        Queue<Integer> q2 = new LinkedList<>();

        for (int i = 0; i < queue1.length; i++) {
            q1.offer(queue1[i]);
            q2.offer(queue2[i]);
        }

        long sum1 = q1.stream().mapToInt(Integer::intValue).sum();
        long sum2 = q2.stream().mapToInt(Integer::intValue).sum();

        while (sum1 != sum2) {
            if (queue1.length + queue2.length + 1 < answer) {
                return -1;
            }
            if (sum1 > sum2) {
                int a = q1.poll();
                q2.offer(a);
                sum1 -= a;
                sum2 += a;
            } else if (sum1 < sum2) {
                int a = q2.poll();
                q1.offer(a);
                sum2 -= a;
                sum1 += a;
            } else {
                return answer;
            }
            answer++;
        }
        return answer;
    }
}

File: Python/프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    q1, q2 = deque(queue1), deque(queue2)
    s1, s2 = sum(q1), sum(q2)
    
    if (s1 + s2) % 2 != 0:
        return -1
    
    while s1 != s2:
        if answer > len(queue1) * 2 + 1:
            return -1
        if s1 < s2:
            cur = q2.popleft()
            q1.append(cur)
            s2 -= cur
            s1 += cur
        else:
            cur = q1.popleft()
            q2.append(cur)
            s1 -= cur
            s2 += cur
        answer += 1
            
    return answer